marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx

```diff
@@ -484,6 +485,9 @@ const MappingStepDetail: React.FC = () => {
       let entProps = resp.data && resp.data[0]["entityModel"].definitions ? resp.data[0]["entityModel"].definitions[rootEntityName].properties : {};
       let entEntityTempData: any = [];
       let nestedEntityProps = extractNestedEntityData(entProps, entEntityTempData);
```

Comment:
Looks good @wooldridge. One minor thing...I'm seeing the initial URI row key is incorrectly set as the last row in the table rather than the first. Seems like the above function call is traversing and incrementing the EntityTableKeyIndex variable before you set uriKey to it. The fix should just be to move the extractNestedEntityData call below line 489 (after uriKey is set and EntityTableKeyIndex is incremented). Hope that makes sense

Reply:
@briantang, thanks for catching that and your suggested fix fixed. I rebased the fixed branch.